service/history/transferQueueStandbyProcessor_test.go

```diff
@@ -56,29 +54,24 @@ type (
 		*require.Assertions
 
 		controller               *gomock.Controller
+		mockResource             *resource.Test
 		mockTxProcessor          *MocktransferQueueProcessor
 		mockReplicationProcessor *MockReplicatorQueueProcessor
 		mockTimerProcessor       *MocktimerQueueProcessor
 		mockDomainCache          *cache.MockDomainCache
+		mockClusterMetadata      *cluster.MockMetadata
+		mockNDCHistoryResender   *xdc.MockNDCHistoryResender
+		mockMatchingClient       *matchingservicetest.MockClient
 
-		mockShardManager        *mocks.ShardManager
-		mockHistoryEngine       *historyEngineImpl
+		mockShard               ShardContext
```

Comment:
maybe it is a good idea to create a mock shard context to be used by all tests

Reply:
I thought about this when modifying the tests. I think there are advantages and disadvantages. A mock shard context will result in lots of duplicated code for creating mocks for things that are already available in `resourceTest` and for asserting expectations, etc. Using `ResourceTest` and the `shardContext` struct can avoid these duplications. However, using mock shard context can make shard context independent from resource during testing. We can create an issue for this and make the decision later.

Reply:
let us do a mock shard, which will improve the code reusability. 
basically doing a initialization with all dependency generated by resourceTest

Reply:
Sure. Let me create a separate pr for this.